### ISSUE TITLE
Arista LANZ (Latency analyzer) input plugin

### ISSUE THREAD

**Issue by timhughes** [influxdata/telegraf#4111]
### Proposal:
I am proposing a new input plugin for monitoring Arista switches. They provide a stream in protobuf and I have written most of the code but I am having some difficulty with the testing which is probably my misunderstanding of how the accumulator works. Code is available at https://github.com/timhughes/telegraf/tree/c4c91ca220109d2f3cba09d5e207e51664ab23bf/plugins/inputs/lanz_consumer

In the test I am creating two endpoints and passing the accumulator to both endpoints. When I input a message `testProtoBufCongestionRecord1` to the first endpoint I get the correct data in the accumulator but when I input the second message `testProtoBufCongestionRecord2` I still only have the first message's  data.

https://www.arista.com/en/solutions/telemetry-analytics

If this question is better off in the community site please let me know and I will move the question there. 

### Use case: [Why is this important (helps with prioritizing requests)]
New plugin allowing more monitoring

**Comment by timhughes** [comment]
I think I have worked out the testing side of things and now have a working multi endpoint consumer service. Currently untested in the real world so  I will put it through it's paces during the week. If someone is able to do a bit of a code review and let me know what needs improving before making a MR it would be appreciated https://github.com/timhughes/telegraf/tree/lanz_consumer_multi_server/plugins/inputs/lanz_consumer

**Comment by littlespace** [comment]
@timhughes , just wanted to see if you have got the plugin working with Arista? also is it gRCP/gNMI or do you need LANZ server to get the plugin work?

**Comment by timhughes** [comment]
We have been using it for a year without any issues. it something i keep meaning to get back and write tests for so that i can get it accepted into the mainline. We have 1 telegraph instance per data centre collecting from the gRPC stream. The code is here https://github.com/timhughes/telegraf/tree/lanz_consumer_multi_server/plugins/inputs/lanz_consumer

**Comment by glinton** [comment]
When you get time to update your pr branch to resolve merge conflicts and add a readme, we can give it another review, seems useful.

**Comment by timhughes** [comment]
I have fixed up the code, rebased and added a readme. PR is #4112 if someone wants to review the code again

**Comment by 612wong** [comment]
@timhughes 

Thank you so much for the lanz_consumer plugins. I found it working very well.
I'm using Grafana to graph the LANZ statistics but end users found they need to lookup what devices are connecting to the interesting interface. As a result, I used snmp plugin to get the ifAlias.

However, influxdb doesn't allow me to join and some said it's good to put the ifAlias into the same measurement in day one.

By looking at the code, I found LANZ output doesn't include the ifAlias.
May I ask if you have alternatives to do this or hacks to the code?

Thanks.
